zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -830,6 +830,29 @@ function ParagraphCtrl ($scope, $rootScope, $route, $window, $routeParams, $loca
       // autocomplete on 'ctrl+.'
       $scope.editor.commands.bindKey('ctrl-.', 'startAutocomplete')
 
+      // Show autocomplete on tab
+      $scope.editor.commands.addCommand({
+        name: 'tabAutocomplete',
+        bindKey: {
+          win: 'tab',
+          mac: 'tab',
+          sender: 'editor|cli'
+        },
+        exec: function(env, args, request) {
+          let iColumnPosition = $scope.editor.selection.getCursor().column
+
+          // If user has pressed tab on first line char or if editor mode is %md, keep existing behavior
+          // If user has pressed tab anywhere in between and editor mode is not %md, show autocomplete
+          if (iColumnPosition && $scope.paragraph.config.editorMode !== 'ace/mode/markdown') {
```

Comment:
@malayhm I agree that one side we don't want to have 10 different completion shortcut keys. But in the other side, just like you made an exception for markdown interpreter, `$scope.paragraph.config.editorMode !== 'ace/mode/markdown'`, `tab` may not appropriate in some cases.

So, do you think there will be no more exception,  except for markdown?